test/integration/users_profile_test.rb

```diff
@@ -18,5 +18,21 @@ def setup
     @user.microposts.paginate(page: 1).each do |micropost|
       assert_match micropost.content, response.body
     end
+    # Micropost Search
+    get user_path(@user), params: {q: {content_cont: "a"}}
+    q = @user.microposts.ransack(content_cont: "a")
+    q.result.paginate(page:1).each do |micropost|
+      assert_match micropost.content, response.body
+    end
+  end
+  
+  test "home profile display" do
+    log_in_as(@user)
+    # Micropost Search
+    get root_path, params: {q: {content_cont: "a"}}
+    q = @user.feed.ransack(content_cont: "a")
+    q.result.paginate(page:1).each do |micropost|
+      assert_match micropost.content, response.body
```

Comment:
HTMLエスケープされてないので文字列比較でマッチしてませんでした。
response.bodyの中身見るとシングルコーテーション`'` とかが `&#39;` とかに変換されてるんですよね。なので失敗してました。

下記のようにすると通ります。

```suggestion
      assert_match CGI.escapeHTML(micropost.content), response.body
```